src/utils/integration/flows.ts

```diff
@@ -50,3 +50,95 @@ export async function listIntegrationFlows(integrationId: string) {
 
   return flows;
 }
+
+export interface LogNode {
+  [index: string]: unknown;
+  timestamp: string;
+  severity: string;
+  message: string;
+}
+
+export interface FetchLogsResult {
+  logs: LogNode[];
+  cursor: string | undefined;
+  executionComplete?: boolean;
+}
+
+export async function listExecutionLogs(executionId: string, nextCursor?: string) {
+  return await gqlRequest({
+    document: gql`
+      query listExecutionLogs($executionId: ID!, $nextCursor: String) {
+        logs(
+          executionResult: $executionId
+          after: $nextCursor
+          orderBy: { field: TIMESTAMP, direction: ASC }
+        ) {
+          edges {
+            node {
+              timestamp
+              severity
+              message
+            }
+            cursor
+          }
+        }
+      }
+    `,
+    variables: {
+      executionId,
+      nextCursor,
+    },
+  });
+}
+
+export interface StepResultNode {
+  [index: string]: unknown;
+  stepName: string;
+  endedAt: string;
+  resultsUrl: string;
+}
+
+export async function listExecutionStepResults(executionId: string, nextCursor?: string) {
+  return await gqlRequest({
+    document: gql`
+      query listExecutionStepResults($executionId: ID!, $nextCursor: String) {
+        executionResult(id: $executionId) {
+          stepResults(after: $nextCursor, orderBy: { field: ENDED_AT, direction: ASC }) {
+            edges {
+              node {
+                stepName
+                endedAt
+                resultsUrl
+              }
+              cursor
+            }
+            totalCount
+          }
+        }
+      }
+    `,
+    variables: {
+      executionId,
+      nextCursor,
+    },
+  });
+}
+
+export async function cniExecutionIsComplete(executionId: string) {
```

Comment:
```suggestion
export async function isCniExecutionComplete(executionId: string) {
```
nit: I'd start with `is`, reads better imo